ssl/ssl_cert.cc

```diff
@@ -153,20 +159,31 @@ UniquePtr<CERT> ssl_cert_dup(CERT *cert) {
     return nullptr;
   }
 
-  if (cert->chain) {
-    ret->chain.reset(sk_CRYPTO_BUFFER_deep_copy(
-        cert->chain.get(), buffer_up_ref, CRYPTO_BUFFER_free));
-    if (!ret->chain) {
-      return nullptr;
-    }
+  ret->cert_private_key_idx = cert->cert_private_key_idx;
+  if (!ssl_cert_check_cert_private_keys_usage(cert) &&
```

Comment:
I think this should be an "or".

Reply:
My bad, good catch. I double checked all usages of `ssl_cert_check_cert_private_keys_usage` again to verify.